modules/apps/portal-security/portal-security-script-management-web/src/main/java/com/liferay/portal/security/script/management/web/internal/portlet/action/GetGroovyScriptUsesMVCResourceCommand.java

```diff
@@ -0,0 +1,127 @@
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Feliphe Marinho
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + ConfigurationAdminPortletKeys.SYSTEM_SETTINGS,
+		"mvc.command.name=/system_settings/get_groovy_script_uses"
+	},
+	service = MVCResourceCommand.class
+)
+public class GetGroovyScriptUsesMVCResourceCommand
+	extends BaseMVCResourceCommand {
+
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_serviceTrackerList = ServiceTrackerListFactory.open(
+			bundleContext, GroovyScriptUsesFactory.class);
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		_serviceTrackerList.close();
+	}
+
+	@Override
+	protected void doServeResource(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		List<GroovyScriptUse> groovyScriptUses = new ArrayList<>();
+
+		if (DBPartition.isPartitionEnabled()) {
+			_companyLocalService.forEachCompany(
+				company -> _addGroovyScriptUses(
+					_serviceTrackerList.iterator(), groovyScriptUses,
+					resourceRequest));
+		}
+		else {
+			_addGroovyScriptUses(
+				_serviceTrackerList.iterator(), groovyScriptUses,
+				resourceRequest);
+		}
+
+		JSONArray jsonArray = _jsonFactory.createJSONArray();
+
+		Comparator<GroovyScriptUse> comparator = Comparator.comparing(
+			GroovyScriptUse::getCompanyWebId);
+
+		groovyScriptUses.sort(
+			comparator.thenComparing(
+				groovyScriptUse -> StringUtil.lowerCase(
+					groovyScriptUse.getSourceName())));
+
+		for (GroovyScriptUse groovyScriptUse : groovyScriptUses) {
+			jsonArray.put(
+				JSONUtil.put(
+					"companyWebId", groovyScriptUse.getCompanyWebId()
+				).put(
+					"sourceName", groovyScriptUse.getSourceName()
+				).put(
+					"sourceURL", groovyScriptUse.getSourceURL()
+				));
+		}
+
+		JSONPortletResponseUtil.writeJSON(
+			resourceRequest, resourceResponse, jsonArray);
+	}
+
+	private void _addGroovyScriptUses(
+		Iterator<GroovyScriptUsesFactory> groovyScriptUseProviderIterator,
+		List<GroovyScriptUse> groovyScriptUses,
+		ResourceRequest resourceRequest) {
+
+		while (groovyScriptUseProviderIterator.hasNext()) {
+			GroovyScriptUsesFactory groovyScriptUsesFactory =
+				groovyScriptUseProviderIterator.next();
+
+			groovyScriptUses.addAll(
+				groovyScriptUsesFactory.create(resourceRequest));
+		}
+	}
+
+	@Reference
+	private CompanyLocalService _companyLocalService;
+
+	@Reference
+	private JSONFactory _jsonFactory;
+
+	private ServiceTrackerList<GroovyScriptUsesFactory> _serviceTrackerList;
```

Comment:
I believe we should keep `_serviceTrackerList` because we are using the same pattern for `Iterator<GroovyScriptUsesFactory> groovyScriptUseProviderIterator` that you asked me to rename to `iterator`.